transformer_engine/pytorch/module/linear.py

```diff
@@ -691,6 +725,11 @@ def forward(
                 else self.noop_cat("weight_tensor", self.weight_names)
             )
 
+            if is_first_microbatch is None:
+                weight1_fp8, weight1_t_fp8 = self.get_fp8_weights_empty_tensors()
+            else:
+                weight1_fp8, weight1_t_fp8 = self.weight1_fp8, self.weight1_t_fp8
```

Comment:
Would this fail for the non-fp8 case since `self.weight1_fp8` and `self.weight1_fp8` don't exist?

Reply:
That's why we had the `if self.fp8` check in the `args +=` part below

Reply:
Correct, fixed this too